### PR TITLE
Update pipelines to use BuildTransportPackage.ps1 

### PR DESCRIPTION
**Pull request by kythant**
Refactor pipelines to use BuildTransportPackage.ps1. Currently the yml and the ps1 does exactly the same thing so it is important that we need to consolidate them. Much of the BuildTransportPackage.ps1 needed refactoring so that the pipeline can use it in steps. Running the BuildTransportPackage.ps1 in the innerloop will run all the steps by default. 

- Updated CI and BuildFoundation to use Powershell Script
- Enable Helix Tests in BuildFoundation.yml if branch is develop
- Move DevCheck to the root directory
- Deleted much of the unused AzurePipelines yml templates or ones that are outdated by the script

### CODE REVIEW

**Comment by dhoehna** [comment]
Does BuildTransportPackages.ps1 still work locally?

File: BuildTransportPackage.ps1 [hunk 1]
```diff
@@ -33,210 +37,257 @@ if(-not (test-path ".nuget\nuget.exe"))
...
-& .\.nuget\nuget.exe restore WindowsAppRuntime.sln -configfile nuget.config
-& .\.nuget\nuget.exe restore "dev\Bootstrap\CS\Microsoft.WindowsAppRuntime.Bootstrap.Net\Microsoft.WindowsAppRuntime.Bootstrap.Net.csproj" -configfile nuget.config
-
-# If the call to restore WindowsAppRuntime_Insights fails check to make sure all Window SDK's from 17760 are installed.
-& .\.nuget\nuget.exe restore "dev\WindowsAppRuntime_Insights\packages.config" -ConfigFile "dev\WindowsAppRuntime_Insights\nuget.config" -PackagesDirectory "dev\WindowsAppRuntime_Insights\packages"
-
+$configurationForMrtAndAnyCPU = "Release"
+$MRTSourcesDirectory = "dev\MRTCore"
 
 $VCToolsInstallDir = . "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -Latest -requires Microsoft.Component.MSBuild -property InstallationPath
```
Comment (dhoehna) [inline BuildTransportPackage.ps1:43]: I had to force the VCTools directory.  MsBuild was taking a different version.

File: BuildTransportPackage.ps1 [hunk 2]
```diff
@@ -33,210 +37,257 @@ if(-not (test-path ".nuget\nuget.exe"))
...
+    }
+    if (($AzureBuildStep -eq "all") -Or ($AzureBuildStep -eq "BuildBinaries")) 
     {
-        write-host "Building WindowsAppRuntime.sln for configuration $configurationToRun and platform:$platformToRun"
-        & $msBuildPath /restore WindowsAppRuntime.sln /p:Configuration=$configurationToRun,Platform=$platformToRun
+        foreach($configurationToRun in $configuration.Split(","))
+        {
+            foreach($platformToRun in $platform.Split(","))
+            {
+                write-host "Building WindowsAppRuntime.sln for configuration $configurationToRun and platform:$platformToRun"
```
Comment (dhoehna) [inline BuildTransportPackage.ps1:112]: The YAML used the /m switch letting the build run multi-threaded.  Locally the build runs into a cyclical dependency issue when building with the /m switch.  If this issue can be figured out the /, can be re-added.

File: BuildTransportPackage.ps1 [hunk 3]
```diff
@@ -33,210 +37,257 @@ if(-not (test-path ".nuget\nuget.exe"))
...
-# Make the foundation transport package.
-nuget pack ".\build\NuSpecs\Microsoft.WindowsAppSDK.Foundation.nuspec" -BasePath $fullNugetPath -OutputDirectory $LocalPackagesPath
-
-# Update the details in eng/version.details.xml
-$packageName = "Microsoft.WindowsAppSDK.Foundation.TransportPackage"
-&"$UpdateVersionDetailsPath" -dependencyName $packageName -dependencyVersion $packageVersion
-
-if(test-path("$env:Build_SourcesDirectory\build\packages\Microsoft.WindowsAppSDK.Foundation.TransportPackage.$packageVersion"))
+        # # Update the details in eng/version.details.xml
+        # $packageName = "Microsoft.WindowsAppSDK.Foundation.TransportPackage"
```
Comment (dhoehna) [inline BuildTransportPackage.ps1:277]: Remove commented code.

File: BuildTransportPackage.ps1 (hunk 2 above)
Comment (kythant) [reply BuildTransportPackage.ps1:112]: Ah that might explain why the old pipelines were faster. I'll workout a solution.

**Comment by kythant** [comment]
> Does BuildTransportPackages.ps1 still work locally?

Yes

**Review by DrusTheAxe — APPROVED** [review]
I love deleting code :-) LGTM

**Pull request merged**